keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlLock.kt

```diff
@@ -0,0 +1,51 @@
+package com.netflix.spinnaker.keel.sql
+
+import com.netflix.spinnaker.keel.sync.Lock
+import org.jooq.DSLContext
+import org.jooq.Field
+import org.jooq.Record
+import org.jooq.Table
+import org.jooq.impl.DSL.field
+import org.jooq.impl.DSL.table
+import org.slf4j.LoggerFactory
+import java.sql.Timestamp
+import java.time.Clock
+import java.time.Duration
+
+class SqlLock(
+  private val jooq: DSLContext,
+  private val clock: Clock
+) : Lock {
+  override fun tryAcquire(name: String, duration: Duration): Boolean =
+    jooq.inTransaction {
+      val now = clock.instant()
+      val limit = now.let(Timestamp::from)
+      val expires = now.plus(duration).let(Timestamp::from)
+      selectOne()
+        .from(LOCK)
+        .where(NAME.eq(name))
+        .forUpdate()
+        .fetch()
+        .intoResultSet()
+        .let {
+          if (it.next()) {
+            update(LOCK)
+              .set(EXPIRES, expires)
+              .where(NAME.eq(name), EXPIRES.lt(limit))
```

Comment:
took me a minute to grok that the update succeeds based on the existing row being expired

thoughts on using the dbs timestamp instead of each client's system clock? (CURRENT_TIMESTAMP is in the jooq DSL)

Reply:
Makes it harder to test